packages/core/src/session.ts

```diff
@@ -376,12 +377,13 @@ class Session {
   /**
    * Update value of the last bookmark.
    * @private
-   * @param {Bookmark} newBookmark - The new bookmark.
+   * @param {string | undefined} newBookmark - The new bookmark.
    * @returns {void}
    */
-  _updateBookmark(newBookmark?: Bookmark): void {
-    if (newBookmark && !newBookmark.isEmpty()) {
-      this._lastBookmark = newBookmark
+  _updateBookmark(newBookmark?: string): void {
+    if (newBookmark !== undefined) {
+      this._bookmarkIn = new Bookmark(newBookmark)
+      this._bookmarkOut = newBookmark
```

Comment:
Actually, the previous `newBookmark` will be always defined. So, it could be changed for `if(!newBookmark.isEmpty())`

Reply:
But what was the `!newBookmark.isEmpty()` there fore before if not to check if the `SUCCESS` message that triggered this update actually contains a bookmark?